lib/message/batch/policy.go

```diff
@@ -218,13 +218,26 @@ func NewPolicy(
 }
 
 //------------------------------------------------------------------------------
+func (p *Policy) Clear() {
+	newParts := make([]types.Part, 0, len(p.parts))
+	for _, part := range p.parts {
+		if part.Metadata().Get("source_msg_lost") != "true" {
+			newParts = append(newParts, part)
+		} else {
+			p.log.Infoln("lost message")
```

Comment:
Can we really loose message?

Reply:
Probably should give a more descriptive message, in the case of azure service bus disconnect we lose lock on a given message so we cannot ack this message even if we process it so it will be redelivered again. Since we will receive this message one more time, we remove it from the batch before processing.

Reply:
Yes, please maybe a better message than the scary `lost message` :-)